src/test/java/tests/FilterEvalTests.java

```diff
@@ -9,19 +9,28 @@
 import java.util.Arrays;
 import java.util.Optional;
 
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 import backend.interfaces.IModel;
 import backend.resource.*;
 import filter.ParseException;
 import filter.Parser;
+import filter.SemanticException;
 import filter.expression.FilterExpression;
 import filter.expression.Qualifier;
+import filter.expression.QualifierType;
 import prefs.Preferences;
 
 public class FilterEvalTests {
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
     private final IModel empty;
+    private static final String SEMANTIC_ERROR_MSG = "Invalid input in filter for"
```

Comment:
Could this field be in one of the exception classes? Otherwise we have to duplicate it here.